File: shared/models/site.py
"""
SQLAlchemy 模型定义 - Site
由代码生成器自动生成 (基于 models.yaml / routes.yaml) - 请勿手动修改
生成时间：2026-05-12 14:56:00
"""

from sqlalchemy import Column, BigInteger, String, Text, Boolean, DateTime, Index

from . import Base  # 使用统一的 Base


class Site(Base):
    """站点模型模型"""
    __tablename__ = 'sites'


    __table_args__ = (
        Index('idx_sites_slug', 'slug'),
        Index('idx_sites_domain', 'domain'),
    )


    id = Column(BigInteger, primary_key=True, autoincrement=True, doc='站点 ID')

    name = Column(String(255), nullable=True, doc='站点名称')

    slug = Column(String(100), unique=True, nullable=True, doc='站点标识')

    domain = Column(String(255), unique=True, nullable=True, doc='主域名')

    additional_domains = Column(Text, nullable=True, doc='附加域名列表（JSON格式）')

    description = Column(Text, nullable=True, doc='站点描述')

    logo_url = Column(String(500), nullable=True, doc='Logo URL')

    favicon_url = Column(String(500), nullable=True, doc='Favicon URL')

    theme = Column(String(100), default='default', doc='主题')

    language = Column(String(10), default='en', doc='语言')

    timezone = Column(String(50), default='UTC', doc='时区')

    settings = Column(Text, nullable=True, doc='站点设置（JSON格式）')


    is_active = Column(Boolean, default=True, doc='是否激活')


    is_default = Column(Boolean, default=False, doc='是否为默认站点')


    created_at = Column(DateTime, doc='创建时间')

    updated_at = Column(DateTime, doc='更新时间')


    def to_dict(self, exclude_sensitive=True):
        """转换为字典

        Args:
            exclude_sensitive: 是否排除敏感字段（密码、密钥、token 等）
        """
        data = {
            'id': self.id,
            'name': self.name,
            'slug': self.slug,
            'domain': self.domain,
            'additional_domains': self.additional_domains,
            'description': self.description,
            'logo_url': self.logo_url,
            'favicon_url': self.favicon_url,
            'theme': self.theme,
            'language': self.language,
            'timezone': self.timezone,
            'settings': self.settings,
            'is_active': self.is_active,
            'is_default': self.is_default,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
        }

        if not exclude_sensitive:
            sensitive_data = {
            }
            data.update(sensitive_data)

        return data

    def __repr__(self):
        """字符串表示"""
        return f'<Site id={self.id}>'

File: frontend/web/src/api/modules/site.ts
/** 多站点管理接口（/api/v3/system/site，system 域） */

import http from '../request'
import type {PageQuery} from '../types'

export interface SiteItem {
  id: number
  name?: string | null
  slug?: string | null
  domain?: string | null
  additional_domains?: string[] | null
  description?: string | null
  logo_url?: string | null
  favicon_url?: string | null
  theme?: string | null
  language?: string | null
  timezone?: string | null
  settings?: Record<string, unknown> | null
  is_active: boolean
  is_default: boolean
  created_at?: string | null
  updated_at?: string | null
}

export interface SiteQuery extends PageQuery {
  is_active?: boolean
}

export interface SitePayload {
  name: string
  slug?: string
  domain?: string
  additional_domains?: string[] | null
  description?: string | null
  logo_url?: string | null
  favicon_url?: string | null
  theme?: string
  language?: string
  timezone?: string
  settings?: Record<string, unknown> | null
  is_active?: boolean
  is_default?: boolean
}

export const siteApi = {
  list: (params?: SiteQuery) => http.page<SiteItem>('/system/site', params),

  create: (data: SitePayload) => http.post<SiteItem>('/system/site', data),

  update: (id: number, data: Partial<SitePayload>) =>
    http.put<SiteItem>(`/system/site/${id}`, data),

  remove: (id: number) => http.delete<null>(`/system/site/${id}`),
}
